aiida/tools/importexport/archive/migrators.py

```diff
@@ -0,0 +1,267 @@
+            raise TypeError('version must be a string')
+
+        if filename and Path(filename).exists() and not force:
+            raise IOError(f'the output path already exists and force=False: {filename}')
+
+        allowed_compressions = ['zip', 'zip-uncompressed', 'tar.gz', 'none']
+        if out_compression not in allowed_compressions:
+            raise ValueError(f'Output compression must be in: {allowed_compressions}')
+
+        MIGRATE_LOGGER.info('Reading archive version')
+        current_version = self._retrieve_version()
+
+        # compute the migration pathway
+        prev_version = current_version
+        pathway: List[str] = []
+        while prev_version != version:
+            if prev_version not in MIGRATE_FUNCTIONS:
+                raise ArchiveMigrationError(f"No migration pathway available for '{current_version}' to '{version}'")
+            if prev_version in pathway:
+                raise ArchiveMigrationError(
+                    f'cyclic migration pathway encountered: {" -> ".join(pathway + [prev_version])}'
+                )
+            pathway.append(prev_version)
+            prev_version = MIGRATE_FUNCTIONS[prev_version][0]
+
+        if not pathway:
+            MIGRATE_LOGGER.info('No migration required')
+            return None
+
+        MIGRATE_LOGGER.info('Migration pathway: %s', ' -> '.join(pathway + [version]))
+
+        # perform migrations
+        if work_dir is not None:
+            migrated_path = self._perform_migration(Path(work_dir), pathway, out_compression, filename)
+        else:
+            with tempfile.TemporaryDirectory() as tmpdirname:
+                migrated_path = self._perform_migration(Path(tmpdirname), pathway, out_compression, filename)
+                MIGRATE_LOGGER.debug('Cleaning temporary folder')
+
+        return migrated_path
+
+    def _perform_migration(
+        self, work_dir: Path, pathway: List[str], out_compression: str, out_path: Optional[Union[str, Path]]
+    ) -> Path:
+        """Perform the migration(s) in the work directory, compress (if necessary),
+        then move to the out_path (if not None).
+        """
+        MIGRATE_LOGGER.info('Extracting archive to work directory')
+
+        extracted = Path(work_dir) / 'extracted'
+        extracted.mkdir(parents=True)
+
+        with get_progress_reporter()(total=1) as progress:
+            callback = create_callback(progress)
+            self._extract_archive(extracted, callback)
+
+        with CacheFolder(extracted) as folder:
+            with get_progress_reporter()(total=len(pathway), desc='Performing migrations: ') as progress:
+                for from_version in pathway:
+                    to_version = MIGRATE_FUNCTIONS[from_version][0]
+                    progress.set_description_str(
+                        f'Performing migrations: {from_version} -> {to_version}', refresh=False
+                    )
+                    progress.update()
```

Comment:
I guess this update meant to update the description of the progress bar (?)

For me it showed 
```
Performing migrations:            0.0%
```
for quite some time - perhaps you want to force a refresh here.

Reply:
fixed in 026b5cb5d755e249407ab2a463529e369523e3c4